torch/csrc/jit/passes/specialize_autogradzero.cpp

```diff
@@ -103,6 +105,27 @@ void specializeAutogradZero(Graph& g) {
       case prim::AutogradZero: {
         state[n->output()] = State::Zero;
       } break;
+      case prim::profile: {
+        // if prim::profile doesn't have an input
+        // it's a counter to keep track how many times
+        // a graph was profiled
+        // otherwise, it sets the state to Unknown
+        // since nothing has been profiled yet.
+        if (n->inputs().size() > 0) {
+          state[n->output()] = State::Unknown;
+        }
+        break;
+      }
+      case prim::BailOut: {
+        if (auto ptt = n->output()->type()->expect<TensorType>()) {
+          state[n->output()] =
+              ptt->autogradZero() ? State::Zero : State::Nonzero;
```

Comment:
autogradZero is an optional boolean, so this first check is just seeing if it is present.

Reply:
If not present, should it get set to Unknown? I don't remember the details of what the states mean in this pass.